pandas/tests/frame/methods/test_reset_index.py

```diff
@@ -183,6 +183,27 @@ def test_reset_index_name(self):
         assert return_value is None
         assert df.index.name is None
 
+    def test_reset_index_rename(self, float_frame):
+        # index
+        rdf = float_frame.reset_index(names="new_name")
+        exp = Series(float_frame.index.values, name="new_name")
+        tm.assert_series_equal(rdf["new_name"], exp)
+
+        # multiindex
+        stacked = float_frame.stack()[::2]
+        stacked = DataFrame({"foo": stacked, "bar": stacked})
+
+        names = ["first", "second"]
+        stacked.index.names = names
+        deleveled = stacked.reset_index()
```

Comment:
I suppose that you do not even need ``deleveled`` here. Indeed, you do not use ``names`` kwarg here. As for assertions, then you probably can do assertions against ``stacked`` instead of ``deleveled``.

Reply:
Here I was trying to test the new renaming feature, by taking advantage of the standard (before this PR) implementation of `reset_index`, where the level names are simply copied over. So `stacked` is the starting point from which `reset_index` creates two DataFrames, which should have exactly the same added columns, except for the names.

I notice now that the step of setting `stacked.index.names` is not relevant, and could/should be skipped.